File: 5-Problem-A-Day/111. Minimum Depth of Binary Tree.java
/**
 * Definition for a binary tree node. public class TreeNode { int val; TreeNode left; TreeNode
 * right; TreeNode(int x) { val = x; } }
 */
class Solution {
    public int minDepth(TreeNode root) {
        if (root == null)
            return 0;
        int left = minDepth(root.left);
        int right = minDepth(root.right);
        if (root.left == null)
            return right + 1;
        if (root.right == null)
            return left + 1;
        return Math.min(left, right) + 1;
    }
}

File: 2021/111. Minimum Depth of Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if (root == NULL) return 0;
        queue<TreeNode*> treeQueue;
        treeQueue.push(root);
        int depth = 0;
        while (!treeQueue.empty()) {
            depth++;
            int size = treeQueue.size();
            for (int i = 0; i < size; i++) {
                root = treeQueue.front();
                treeQueue.pop();
                if (root->left == NULL && root->right == NULL) return depth;
                if (root->left != NULL) {
                    treeQueue.push(root->left);
                }
                if (root->right != NULL) {
                    treeQueue.push(root->right);
                }
            }
        }
        return 0;
    }
};